libfwupdplugin/fu-dummy-efivars.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#define G_LOG_DOMAIN "FuDummyEfivars"
+
+#include "config.h"
+
+#include "fu-dummy-efivars.h"
+
+typedef struct {
+	gchar *guid;
+	gchar *name;
+	guint32 attr;
+	GByteArray *buf;
+} FuDummyEfivarsKey;
+
+struct _FuDummyEfivars {
+	FuEfivars parent_instance;
+	GPtrArray *keys; /* of FuDummyEfivarsKey */
+};
+
+G_DEFINE_TYPE(FuDummyEfivars, fu_dummy_efivars, FU_TYPE_EFIVARS)
+
+static void
+fu_dummy_efivars_key_free(FuDummyEfivarsKey *key)
+{
+	g_free(key->guid);
+	g_free(key->name);
+	g_byte_array_unref(key->buf);
+	g_free(key);
+}
+
+static gboolean
+fu_dummy_efivars_supported(FuEfivars *efivars, GError **error)
+{
+	return TRUE;
+}
```

Comment:
How about guarding the dummy interface to ONLY work when an env variable is set?

Reply:
We've also using this in the installed tests -- we could do something like `g_setenv("FWUPD_EFIVARS", "dummy", TRUE);` and load that in the context if that's set?

Reply:
Yes; that's exactly what I was thinking.  It will make sure that the host never impacts the tests too.

Reply:
I had a better idea; just compile the fu-dummy-efivars.c into the test suite -- it's the only place we need it. Then we can rip out a **lot** of complexity!